wallet/src/Dscp/Wallet/KeyStorage.hs

```diff
@@ -0,0 +1,67 @@
+module Dscp.Wallet.KeyStorage
+       ( getAccounts
+       , addAccount
+       ) where
+
+import Data.Aeson (eitherDecode, encode)
+import Data.Aeson.TH (defaultOptions, deriveJSON)
+import Dscp.Core (mkAddr)
+import Dscp.Util.Aeson (Versioned(..))
+import System.Directory (doesFileExist)
+import System.FileLock (SharedExclusive (..), withFileLock)
+
+import qualified Data.ByteString.Lazy as LBS
+
+import Dscp.Wallet.Face
+
+data Storage = Storage
+    { accounts :: [StorageAccount]
+    }
+
+data StorageAccount = StorageAccount
+    { secretKey :: Encrypted SecretKey
+    , publicKey :: PublicKey
+    }
+
+deriveJSON defaultOptions ''Storage
+deriveJSON defaultOptions ''StorageAccount
+
+storageFileName :: FilePath
+storageFileName = "run/tmp/wallet.json"
+
+storageLockFileName :: FilePath
+storageLockFileName = "run/tmp/wallet.json.lock"
+
+getAccounts :: IO [Account]
+getAccounts = do
+    storage <- readStorage
+    return $ toAccount <$> accounts storage
+  where
+    toAccount StorageAccount{..} = Account
+        { accountSecretKey = secretKey
+        , accountPublicKey = publicKey
+        , accountAddress = mkAddr publicKey
+        }
+
+addAccount :: Account -> IO ()
+addAccount account = modifyStorage (\(Storage accs) -> Storage $ accs ++ [fromAccount account])
+  where
+    fromAccount Account{..} = StorageAccount
+        { secretKey = accountSecretKey
+        , publicKey = accountPublicKey
+        }
+
+readOrCreateStorage :: IO Storage
+readOrCreateStorage = do
+    exists <- doesFileExist storageFileName
+    if exists
+        then LBS.readFile storageFileName >>= either fail (\(Versioned a) -> return a) . eitherDecode
```

Comment:
I don't know whether any exception handling practices are enforced in DSCP, but I personally dislike `fail` (and it was prohibited by the guidelines we had in `cardano-sl`).
It's recommended to create a custom exception type and throw it instead.